clients/src/main/java/org/apache/kafka/common/record/MemoryRecordsBuilder.java

```diff
@@ -154,13 +154,18 @@ public void close() {
             throw new KafkaException(e);
         }
 
-        if (compressionType != CompressionType.NONE)
-            writerCompressedWrapperHeader();
+        if (numRecords == 0L) {
+            buffer().position(initPos);
```

Comment:
Out of curiosity, why do we do this here? In the normal case, we only update the position of the duplicate buffer, it seems.